In [29]:
import pandas as pd

In [30]:
# products_cl.csv
products_cl = pd.read_csv("/Users/sreemalathi/Documents/Bootcamp/Python/Pandas/Eniac Project/cl_files/products_cl.csv")

In [31]:
products_cl['sub_category'] = ""

In [32]:
products_cl["product_condition"] = "new"
products_cl.loc[
    products_cl["name"].str.contains("open|like new|second hand|abierto", case=False, na=False),
    "product_condition"
] = "refurbished"

In [33]:

# Verify
products_cl["product_condition"].value_counts()
# 1602/9992 16% refurbished products 

product_condition
new            8390
refurbished    1602
Name: count, dtype: int64

In [34]:
type_map = {
    "11865403": "iPhone/iPad Case",
    "12635403": "iPhone/iPad Case",
    "12575403": "iPhone/iPad Case",
    "12175397": "NAS / Network Storage",
    "1404": "NAS / Network Storage",
    "1280": "NAS / Network Storage",
    "11935397": "Hard Drive",
    "12655397": "Hard Drive",
    "11905404": "Smart Home / Gadgets",
    "5404": "Smart Home / Gadgets",
    "13835403": "MacBook Case",
    "5,74E+15": "Mac Desktop",
    "118692158": "Mac Desktop",
    "2,16E+11": "Mac Desktop",
    "21632158": "Mac Desktop",
    "5,43E+15": "Mac Desktop",
    "5,72E+15": "Mac Desktop",
    "1364": "RAM / Memory",
    "12585395": "Adapters / Hubs",
    "5395": "Adapters / Hubs",
    "14365395": "Adapters / Hubs",
    "1276": "Adapters / Hubs",
    "1296": "Monitor",
    "1325": "Cable",
    "1230": "Cable",
    "5384": "Headphones",
    "1433": "SSD",
    "12215397": "SSD",
    "12755395": "SSD",
    "5398": "Speaker",
    "1,02E+12": "Mac Laptop",
    "2158": "Mac Laptop",
    "5,39E+11": "Mac Laptop",
    "2,17E+11": "Mac Laptop",
    "57445397": "USB / Memory",
    "42945397": "USB / Memory",
    "1,44E+11": "Repair Service",
    "20642062": "Repair Service",
    "1334": "Networking",
    "2449": "Apple Watch Accessory",
    "24215399": "Apple Watch Accessory",
    "2434": "Apple Watch Accessory",
    "1229": "Stylus",
    "12995397": "Dock",
    "1515": "Power Bank / Charger",
    "13615399": "Power Bank / Charger",
    "5,49E+11": "Power Bank / Charger",
    "13005399": "Power Bank / Charger",
    "5399": "Power Bank / Charger",
    "1405": "Graphics Tablet",
    "101781405": "Graphics Tablet",
    "13555403": "Screen Protector",
    "14035403": "Screen Protector",
    "1216": "Mount / Stand",
    "8696": "Mount / Stand",
    "5720": "Mount / Stand",
    "12355400": "Mount / Stand",
    "24885185": "Apple Watch",
    "24895185": "Apple Watch",
    "21485407": "Replacement Part",
    "12645406": "Replacement Part",
    "14305406": "Replacement Part",
    "10142": "Replacement Part",
    "21535407": "Replacement Part",
    "1392": "Backpack",
    "10230": "Backpack",
    "11821715": "Music Player",
    "9094": "Security Camera",
    "51601716": "iPhone",
    "85651716": "iPhone",
    "85641716": "iPhone",
    "1716": "iPhone",
    "24821716": "iPhone",
    "24811716": "iPhone",
    "113281716": "iPhone",
    "113291716": "iPhone",
    "5405": "Sport Armband",
    "13855401": "Keyboard",
    "12141714": "iPad",
    "106431714": "iPad",
    "51861714": "iPad",
    "1714": "iPad",
    "51871714": "iPad",
    "13621714": "iPad",
    "42931714": "iPad",
    "24861714": "iPad",
    "1416": "Software",
    "1387": "Mouse / Trackpad",
    "5403": "Case",
    "1375": "Microphone",
    "54864259": "Remote Control",
    "4259": "Apple TV",
    "12285400": "Other",
    "12085400": "Other",
}

In [35]:
products_cl["sub_category"] = products_cl["type"].map(type_map)
products_cl["sub_category"] = products_cl["sub_category"].fillna("Other")

In [36]:
laptop_mask = (products_cl["type"] == "1282") & (products_cl["name"].str.contains("MacBook", case=False, na=False))
desktop_mask = (products_cl["type"] == "1282") & (products_cl["name"].str.contains("iMac|Mac mini|Mac Pro", case=False, na=False))

products_cl.loc[laptop_mask, "sub_category"] = "Mac Laptop"
products_cl.loc[desktop_mask, "sub_category"] = "Mac Desktop"

products_cl.loc[(products_cl["type"] == "1282") & ~laptop_mask & ~desktop_mask, "name"]

6792    Second hand - Apple LED Cinema Display 27 "Mid...
6848          Second hand - Apple LED Cinema Display 27 "
6849                        Apple LED Cinema Display 24 "
9593          Second hand - Apple LED Cinema Display 24 "
Name: name, dtype: str

In [37]:
products_cl.loc[
    (products_cl["type"] == "1282") & ~laptop_mask & ~desktop_mask,
    "sub_category"
] = "Monitor"

In [38]:
category_map = {
    # Apple Devices
    "iPhone": "Apple Devices",
    "iPad": "Apple Devices",
    "Mac Laptop": "Apple Devices",
    "Mac Desktop": "Apple Devices",
    "Apple Watch": "Apple Devices",
    "Apple TV": "Apple Devices",
    "Music Player": "Apple Devices",
    "Monitor": "Apple Devices",

    # Protection & Cases
    "MacBook Case": "Protection & Cases",
    "iPhone/iPad Case": "Protection & Cases",
    "Screen Protector": "Protection & Cases",
    "Case": "Protection & Cases",

    # Audio
    "Headphones": "Audio",
    "Speaker": "Audio",
    "Microphone": "Audio",

    # Storage & Memory
    "RAM / Memory": "Storage & Memory",
    "SSD": "Storage & Memory",
    "Hard Drive": "Storage & Memory",
    "NAS / Network Storage": "Storage & Memory",
    "USB / Memory": "Storage & Memory",

    # Accessories
    "Mount / Stand": "Accessories",
    "Apple Watch Accessory": "Accessories",
    "Backpack": "Accessories",
    "Sport Armband": "Accessories",

    # Connectivity, Power & Charging
    "Power Bank / Charger": "Connectivity, Power & Charging",
    "Cable": "Connectivity, Power & Charging",
    "Adapters / Hubs": "Connectivity, Power & Charging",
    "Dock": "Connectivity, Power & Charging",

    # Input Devices
    "Keyboard": "Input Devices",
    "Mouse / Trackpad": "Input Devices",
    "Stylus": "Input Devices",
    "Graphics Tablet": "Input Devices",
    "Remote Control": "Input Devices",

    # Networking & Smart Home
    "Networking": "Networking & Smart Home",
    "Smart Home / Gadgets": "Networking & Smart Home",
    "Security Camera": "Networking & Smart Home",

    # Services & Parts
    "Replacement Part": "Services & Parts",
    "Repair Service": "Services & Parts",

    # Software
    "Software": "Software",

    # Other
    "Other": "Other",
}

products_cl["category"] = products_cl["sub_category"].map(category_map)
products_cl["category"].value_counts()

category
Storage & Memory                  2401
Apple Devices                     1869
Protection & Cases                1826
Other                              951
Connectivity, Power & Charging     870
Networking & Smart Home            631
Accessories                        498
Audio                              346
Services & Parts                   323
Input Devices                      253
Software                            24
Name: count, dtype: int64

# Note
- Product_condition: new (8,390) / refurbished (1,602)
- sub_category: 39 categories, ~90.5% covered, rest in "Other"
- category: 10 broad groups for high-level analysis
other - type1298 has 783 rows but not classifed as the data was mixed/messy

In [39]:
products_cl.to_csv("products_category.csv", index=False)